sources/packages/web/src/store/modules/student/student.ts

```diff
@@ -10,19 +12,25 @@ export class StudentProfile {
   gender?: string;
   givenName?: string;
   middleName?: string;
-  validSIN?: boolean | null;
+  validSIN?: boolean;
+}
+
+export class SINValidStatus {
+  sinStatus?: SinStatusEnum;
+  severity?: string;
+  message?: string;
 }
 
 export class StudentState {
   profile: StudentProfile = new StudentProfile();
   hasStudentAccount = false;
-  validSIN = false;
+  sinValidStatus: SINValidStatus = new SINValidStatus();

```

Comment:
@henryoforeh-aot We could have used the SinStatusEnum here instead of SINValidStatus. Am I missing something ?

Reply:
I created class called SINValidStatus to encapsulate the following attributes sinstatus, severity and message. This was done to save the state of those attributes and reduce logic in the CheckValidSINBanner.vue component.